src/libponyc/pkg/buildflagset.c

```diff
@@ -534,6 +534,29 @@ bool define_build_flag(const char* name)
 }
 
 
+bool remove_build_flags(const char* flags[])
+{
+  pony_assert(flags != NULL);
+
+  if(_user_flags == NULL)
+  {
+    // Initialise flags table.
+    _user_flags = POOL_ALLOC(flagtab_t);
+    flagtab_init(_user_flags, 8);
+  }
+
+  size_t removed = 0;
+  for(const char** next = flags; *next != NULL; next += 1)
+  {
+    flag_t f1 = {stringtab(*next), false};
+    if(flagtab_remove(_user_flags, &f1) != NULL)
```

Comment:
You need to save the return `flag_t*` from `flagtab_remove` and then free it via `flag_free` if it is not `NULL` to avoid a memory leak.

Reply:
Good catch, fixing.